pkg/operator/controller/ingress/controller.go

```diff
@@ -497,28 +498,31 @@ func (r *reconciler) ensureIngressDeleted(ingress *operatorv1.IngressController)
 }
 
 // ensureIngressController ensures all necessary router resources exist for a given ingresscontroller.
-func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, dnsConfig *configv1.DNS, infraConfig *configv1.Infrastructure, ingressConfig *configv1.Ingress, apiConfig *configv1.APIServer) error {
+// Any error values are collected into an Aggregate error value.  This error value is returned along
+// with a duration value that indicates, if it is non-zero, that the operator should reconcile the
+// ingresscontroller again after that duration has elapsed.
+func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, dnsConfig *configv1.DNS, infraConfig *configv1.Infrastructure, ingressConfig *configv1.Ingress, apiConfig *configv1.APIServer) (time.Duration, error) {
```

Comment:
What about instead using a semantic error pattern? Like returning a RetryableError which embeds the retry context, and on the call site checking the error type

Reply:
If we do that, `ensureIngressController` will need to wrap its aggregate error value with a `RetryableError` with the minimum of the retry period of any `RetryableError` in `errs` (this could be done with some `NewMaybeRetryableAggregate` function), so it might not reduce complexity as much as you are hoping.

Reply:
I pushed an update that uses this approach.

Reply:
I certainly hadn't considered aggregates... looks like you went for it anyway, so guess we'll figure it out!